unity/CAVE/Assets/W3D_Translator/CLI.cs

```
@@ -184,9 +187,57 @@ static void Main()
         return;
     }
 
+    // Convert Story.ObjectRoot to a dictionary of {name: GameObject} pairs
+    static Dictionary<string, GameObject> TranslateGameObjects(
+        List<W3D.Object> objectList, GameObject story
+    ) {
+        Dictionary<string, GameObject> gameObjects = new Dictionary<string, GameObject>();
+        /** Object
+            name: gameObject.name
+            Visible: gameObject.active
+            Color: gameObject.[content].color
+            Lighting: TODO (76)
+            ClickThrough: TODO (76)
+            AroundSelfAxis: TODO (76)
+            Scale: gameObject.localScale (set in Placement.SetTransform)
+        */
+        foreach (W3D.Object xml in objectList)
+        {   
+            GameObject gameObject = new GameObject();
+            gameObject.name = xml.Name;
+            xml.Placement.SetTransform(gameObject.transform, xml.Scale, story.transform);
+            
+            // TODO 74: LinkRoot.Link -> Add a VRCanvas
+            if(xml.LinkRoot is not null) {}
+    
+            // Add Content component(s)
+            switch(xml.Content.ContentData) {
+                case(Text textContent):
+                    textContent.GenerateTMP(gameObject, Xml.ConvertColor(xml.ColorString));
+                    break;
+                case(Image imageContent):
+                    break; // TODO (65)
+                case(StereoImage stereoImageContent):
+                    break; // TODO (66)
+                case(Model modelContent):
+                    break; // TODO (67)
+                case(W3D.Light lightContent):
+                    break; // TODO (68)
+                case(W3D.ParticleSystem particleSystemContent):
+                    break; // TODO (69)
+                default: break;
```

I'd throw an error here

---

Will do!